packages/react/src/components/Layer/Layer.notification.ts

```diff
@@ -4,29 +4,31 @@ let _defaultHostSelector: string | undefined;
 
 /**
  * Register a layer for a given host id
- * @param hostId Id of the layer host
+ * @param hostId Id or Node of the layer host
  * @param layer Layer instance
  */
-export function registerLayer(hostId: string, callback: () => void) {
-  if (!_layersByHostId[hostId]) {
-    _layersByHostId[hostId] = [];
+export function registerLayer(hostId: string | Node, callback: () => void) {
+  const nodeName = typeof hostId === 'string' ? hostId : (hostId as Node).nodeName;
```

Comment:
`nodeName` is not unique; it's actually likely to be `DIV` for most layers.

Reply:
In the context I was describing in the attached issue the purpose is to send a unique custom element. (not any random node from the light-dom, because it is already covered by getting it by id).
I see though that it is not robust enough because it can be misused.

Do you have any suggestion? Should we rather open up for sending a type ShadowRoot, so if it receives a string we search that id through the light-dom (as per today), and if it receives a shadowRoot we get the first Node in that shadow?